library/src/cache/patch_manager.rs

```diff
@@ -0,0 +1,683 @@
+use super::{disk_io, PatchInfo};
+use anyhow::{bail, Context, Ok, Result};
+use core::fmt::Debug;
+use serde::{Deserialize, Serialize};
+use std::path::{Path, PathBuf};
+
+#[cfg(test)]
+use mockall::automock;
+#[cfg(test)]
+use tempdir::TempDir;
+
+// https://stackoverflow.com/questions/67087597/is-it-possible-to-use-rusts-log-info-for-tests
+#[cfg(test)]
+use std::{println as info, println as error}; // Workaround to use println! for logs.
+
+const PATCHES_DIR_NAME: &str = "patches";
+const PATCHES_STATE_FILE_NAME: &str = "patches_state.json";
+
+#[derive(Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq, Hash)]
+struct PatchMetadata {
```

Comment:
So is the patch just one file or is it a directory?

Reply:
The updater's directory structure now looks like:

```
shorebird_updater/
  state.json
  patches_state.json
  patches/
    1/
       dlc.vmcode
    2/
       dlc.vmcode
```

Is that helpful?

Reply:
Which part of that does PatchMetadata refer to?

Reply:
And will you have to change it when the next patch you make to this Rust code will be to support multiple vmcode files for iOS? 🤣

Reply:
This code makes me wish that .dmg files (or some other seamless container, maybe .tar?) worked well on iOS. So that you could have multiple files appear as one file on disk.

Reply:
Ah, I understand now. `PatchMetadata` comes baked with the assumption that the patch is a single file. We'll need to change this to a map of file names to sizes (or a collection of a yet-to-be-defined `PatchFile` type) down the road.